examples/3d/debug_gltf.rs

```diff
@@ -0,0 +1,112 @@
+use bevy::{
+    gltf::{Gltf, GltfNode},
+    prelude::*,
+};
+
+fn main() {
+    App::new()
+        .insert_resource(AmbientLight {
+            color: Color::WHITE,
+            brightness: 1.0 / 5.0f32,
+        })
+        .add_plugins(DefaultPlugins)
+        .add_startup_system(setup)
+        .insert_resource(PrintDone::default())
+        .insert_resource(AssertDone::default())
+        .add_system(spawn_and_print_gltf_objects)
+        .add_system(assert_correctness)
+        .run();
+}
+
+fn setup(mut commands: Commands, assets: Res<AssetServer>) {
+    // Gltf can be packed in a more compact format called glb.
+    // Unlike Gltf this format isn't human readable
+    // so being able to print the loaded structure is extra valuable
+    let handle: Handle<Gltf> = assets.load("models/AlienCake/alien.glb");
+    commands.insert_resource(handle);
+
+    commands.spawn_bundle(PerspectiveCameraBundle {
+        transform: Transform::from_xyz(0.5, 1.0, -2.0)
+            .looking_at(Vec3::new(0.0, 0.3, 0.0), Vec3::Y),
+        ..Default::default()
+    });
+    const HALF_SIZE: f32 = 1.0;
+    commands.spawn_bundle(DirectionalLightBundle {
+        directional_light: DirectionalLight {
+            shadow_projection: OrthographicProjection {
+                left: -HALF_SIZE,
+                right: HALF_SIZE,
+                bottom: -HALF_SIZE,
+                top: HALF_SIZE,
+                near: -10.0 * HALF_SIZE,
+                far: 10.0 * HALF_SIZE,
+                ..Default::default()
+            },
+            shadows_enabled: true,
+            ..Default::default()
+        },
+        transform: Transform::from_xyz(-0.7, 1.7, -2.0)
+            .looking_at(Vec3::new(0.0, 0.3, 0.0), Vec3::Y),
+        ..Default::default()
+    });
+}
+
+fn spawn_and_print_gltf_objects(
+    mut commands: Commands,
+    mut done: ResMut<PrintDone>,
+    gltf_handle: Res<Handle<Gltf>>,
+    assets_gltf: Res<Assets<Gltf>>,
+    assets_gltfnode: Res<Assets<GltfNode>>,
+) {
+    if done.0 {
+        return;
+    }
+
+    // if the GLTF has loaded, we can navigate its contents
+    if let Some(gltf) = assets_gltf.get(gltf_handle.clone()) {
```

Comment:
Nice to see an example use the `Gltf` struct instead of loading directly using one of the label! This is something that's often confusing to user, when to load without a label, with `#Scene0` or with `#Mesh0/Primitive0`.

What would you think of adding a `info!("The GLTF files contains the following assets: {:#?}, gltf);`. I'm not sure it's very readable but it's still interesting, what do you think?

Reply:
The output looks like this:
```
The GLTF file contains the following assets: Gltf {
    scenes: [
        StrongHandle<Scene>(AssetPathId(AssetPathId(SourcePathId(14247583744215955211), LabelId(17908982665524773019)))),
    ],
    named_scenes: {
        "Scene": StrongHandle<Scene>(AssetPathId(AssetPathId(SourcePathId(14247583744215955211), LabelId(17908982665524773019)))),
    },
    meshes: [
        StrongHandle<GltfMesh>(AssetPathId(AssetPathId(SourcePathId(14247583744215955211), LabelId(4338101777398872398)))),
        StrongHandle<GltfMesh>(AssetPathId(AssetPathId(SourcePathId(14247583744215955211), LabelId(10650818211917448613)))),
        StrongHandle<GltfMesh>(AssetPathId(AssetPathId(SourcePathId(14247583744215955211), LabelId(16514377650528378588)))),
        StrongHandle<GltfMesh>(AssetPathId(AssetPathId(SourcePathId(14247583744215955211), LabelId(3898330275780202977)))),
        StrongHandle<GltfMesh>(AssetPathId(AssetPathId(SourcePathId(14247583744215955211), LabelId(15634264708557667658)))),
        StrongHandle<GltfMesh>(AssetPathId(AssetPathId(SourcePathId(14247583744215955211), LabelId(10754852173108149575)))),
    ],
    named_meshes: {
        "Mesh legLeft": StrongHandle<GltfMesh>(AssetPathId(AssetPathId(SourcePathId(14247583744215955211), LabelId(15634264708557667658)))),
        "Mesh legRight": StrongHandle<GltfMesh>(AssetPathId(AssetPathId(SourcePathId(14247583744215955211), LabelId(10754852173108149575)))),
        "Mesh head": StrongHandle<GltfMesh>(AssetPathId(AssetPathId(SourcePathId(14247583744215955211), LabelId(3898330275780202977)))),
        "Mesh armRight": StrongHandle<GltfMesh>(AssetPathId(AssetPathId(SourcePathId(14247583744215955211), LabelId(10650818211917448613)))),
        "Mesh armLeft": StrongHandle<GltfMesh>(AssetPathId(AssetPathId(SourcePathId(14247583744215955211), LabelId(4338101777398872398)))),
        "Mesh body": StrongHandle<GltfMesh>(AssetPathId(AssetPathId(SourcePathId(14247583744215955211), LabelId(16514377650528378588)))),
    },
    materials: [
        StrongHandle<StandardMaterial>(AssetPathId(AssetPathId(SourcePathId(14247583744215955211), LabelId(3443497372379138987)))),
        StrongHandle<StandardMaterial>(AssetPathId(AssetPathId(SourcePathId(14247583744215955211), LabelId(13375458626705325195)))),
        StrongHandle<StandardMaterial>(AssetPathId(AssetPathId(SourcePathId(14247583744215955211), LabelId(3330348954263395478)))),
        StrongHandle<StandardMaterial>(AssetPathId(AssetPathId(SourcePathId(14247583744215955211), LabelId(114420222998594830)))),
        StrongHandle<StandardMaterial>(AssetPathId(AssetPathId(SourcePathId(14247583744215955211), LabelId(17458303146753488220)))),
    ],
    named_materials: {
        "crystal": StrongHandle<StandardMaterial>(AssetPathId(AssetPathId(SourcePathId(14247583744215955211), LabelId(3443497372379138987)))),
        "metal": StrongHandle<StandardMaterial>(AssetPathId(AssetPathId(SourcePathId(14247583744215955211), LabelId(13375458626705325195)))),
        "metalRed": StrongHandle<StandardMaterial>(AssetPathId(AssetPathId(SourcePathId(14247583744215955211), LabelId(3330348954263395478)))),
        "dark": StrongHandle<StandardMaterial>(AssetPathId(AssetPathId(SourcePathId(14247583744215955211), LabelId(17458303146753488220)))),
        "metalDark": StrongHandle<StandardMaterial>(AssetPathId(AssetPathId(SourcePathId(14247583744215955211), LabelId(114420222998594830)))),
    },
    nodes: [
        StrongHandle<GltfNode>(AssetPathId(AssetPathId(SourcePathId(14247583744215955211), LabelId(17756025687278476026)))),
        StrongHandle<GltfNode>(AssetPathId(AssetPathId(SourcePathId(14247583744215955211), LabelId(14260767632020858270)))),
        StrongHandle<GltfNode>(AssetPathId(AssetPathId(SourcePathId(14247583744215955211), LabelId(10061463035338297464)))),
        StrongHandle<GltfNode>(AssetPathId(AssetPathId(SourcePathId(14247583744215955211), LabelId(13405862217271904705)))),
        StrongHandle<GltfNode>(AssetPathId(AssetPathId(SourcePathId(14247583744215955211), LabelId(4105411459194179812)))),
        StrongHandle<GltfNode>(AssetPathId(AssetPathId(SourcePathId(14247583744215955211), LabelId(922133183154009395)))),
        StrongHandle<GltfNode>(AssetPathId(AssetPathId(SourcePathId(14247583744215955211), LabelId(16788655570146856287)))),
        StrongHandle<GltfNode>(AssetPathId(AssetPathId(SourcePathId(14247583744215955211), LabelId(12708891262738597888)))),
    ],
    named_nodes: {
        "head": StrongHandle<GltfNode>(AssetPathId(AssetPathId(SourcePathId(14247583744215955211), LabelId(13405862217271904705)))),
        "armRight": StrongHandle<GltfNode>(AssetPathId(AssetPathId(SourcePathId(14247583744215955211), LabelId(14260767632020858270)))),
        "alien": StrongHandle<GltfNode>(AssetPathId(AssetPathId(SourcePathId(14247583744215955211), LabelId(16788655570146856287)))),
        "legRight": StrongHandle<GltfNode>(AssetPathId(AssetPathId(SourcePathId(14247583744215955211), LabelId(922133183154009395)))),
        "armLeft": StrongHandle<GltfNode>(AssetPathId(AssetPathId(SourcePathId(14247583744215955211), LabelId(17756025687278476026)))),
        "body": StrongHandle<GltfNode>(AssetPathId(AssetPathId(SourcePathId(14247583744215955211), LabelId(10061463035338297464)))),
        "tmpParent": StrongHandle<GltfNode>(AssetPathId(AssetPathId(SourcePathId(14247583744215955211), LabelId(12708891262738597888)))),
        "legLeft": StrongHandle<GltfNode>(AssetPathId(AssetPathId(SourcePathId(14247583744215955211), LabelId(4105411459194179812)))),
    },
    default_scene: Some(
        StrongHandle<Scene>(AssetPathId(AssetPathId(SourcePathId(14247583744215955211), LabelId(17908982665524773019)))),
    ),
    scene_to_nodes: {
        StrongHandle<Scene>(AssetPathId(AssetPathId(SourcePathId(14247583744215955211), LabelId(17908982665524773019)))): [
            StrongHandle<GltfNode>(AssetPathId(AssetPathId(SourcePathId(14247583744215955211), LabelId(12708891262738597888)))),
        ],
    },
}
```
I'm not sure this adds much value, but I'm willing to be convinced of the opposite.